From: "Rémi Bernon" <rbernon@codeweavers.com>
Subject: [PATCH v2 2/2] services: Terminate all service processes on shutdown.
Message-Id: <20211110092646.419531-2-rbernon@codeweavers.com>
Date: Wed, 10 Nov 2021 10:26:46 +0100
In-Reply-To: <20211110092646.419531-1-rbernon@codeweavers.com>
References: <20211110092646.419531-1-rbernon@codeweavers.com>

Instead of waiting for wineserver to kills them on shutdown, after a
2s delay.

This saves another ~2s on the prefix shutdown time, as measured with:

  time bash -c 'wine cmd /c exit |& tee /dev/null'

Signed-off-by: Rémi Bernon <rbernon@codeweavers.com>
---
 programs/services/services.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/programs/services/services.c b/programs/services/services.c
index 39cd9ee4ffa..d969cf81ab9 100644
--- a/programs/services/services.c
+++ b/programs/services/services.c
@@ -520,6 +520,7 @@ static void scmdatabase_wait_terminate(struct scmdatabase *db)
     {
         struct process_entry *process = grab_process(LIST_ENTRY(ptr, struct process_entry, entry));
 
+        process_terminate(process);
         scmdatabase_unlock(db);
         WaitForSingleObject(process->process, INFINITE);
         scmdatabase_lock(db);

-- 
2.33.1

